server/src/main/java/org/tctalent/server/model/db/DuolingoCoupon.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2024 Talent Beyond Boundaries.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation, either version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+package org.tctalent.server.model.db;
+
+import java.time.LocalDateTime;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.lang.Nullable;
+
+@Getter
+@Setter
+@Entity
+@Table(name = "duolingo_coupon")
+@SequenceGenerator(name = "seq_gen", sequenceName = "coupon_id_seq", allocationSize = 1)
+public class DuolingoCoupon {
```

Comment:
Can you add some documentation to this class. Describe what a Duolingo coupon is - and then describe each field.